File: api/src/models/Comment.js
import mongoose from 'mongoose';

const commentSchema = new mongoose.Schema({
  user: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  content: {
    type: String,
    required: true
  },
  upvote: {
    type: Number,
    default: 0
  },
  downvote: {
    type: Number,
    default: 0
  },
  reply: {
    type: String
  },
  postId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Post',
    required: true
  },
  created_at: {
    type: Date,
    default: Date.now
  }
});

const Comment = mongoose.model('Comment', commentSchema);

export default Comment;

File: course_hub/lib/models/comment.dart
class Comment {
  String id;
  String username;
  String content;
  int upvote;
  int downvote;

  Comment({
    required this.id,
    required this.username,
    required this.content,
    required this.upvote,
    required this.downvote,
  });

  factory Comment.fromJson(Map<String, dynamic> json) {
    return Comment(
      id: json['_id'] ?? '',
      username: json['user']["username"] ?? '',
      content: json['content'] ?? '',
      upvote: json['upvote'] ?? 0,
      downvote: json['downvote'] ?? 0,
    );
  }
}
